Library/Homebrew/utils/github.rb

```diff
@@ -357,18 +357,18 @@ def self.get_artifact_url(workflow_array)
     run_id = check_suite.last["workflowRun"]["databaseId"]
     artifacts = API.open_rest("#{API_URL}/repos/#{user}/#{repo}/actions/runs/#{run_id}/artifacts", scopes:)
 
-    artifact = artifacts["artifacts"].select do |art|
-      art["name"] == artifact_name
+    matching_artifacts = artifacts["artifacts"].select do |art|
+      File.fnmatch?(artifact_pattern, art["name"], File::FNM_EXTGLOB)
     end
 
-    if artifact.empty?
+    if matching_artifacts.empty?
       raise API::Error, <<~EOS
-        No artifact with the name `#{artifact_name}` was found!
+        No artifacts with the pattern `#{artifact_pattern}` were found!
           #{Formatter.url check_suite.last["workflowRun"]["url"]}
       EOS
     end
 
-    artifact.last["archive_download_url"]
+    matching_artifacts.map { |art| art["archive_download_url"] }
```

Comment:
It's technically possible to have two artifacts with the same name still, even under v4 (there's a retry bug that causes that).

The `artifact.last` was intentional to avoid that, so we probably want to add logic that groups by name and returns the last download URL for each.

Reply:
Added in 9ad60fe437c0d524836147086c528cc86affc51f